### PR TITLE
303 - update the buttons demo page with a customization bottom sheet

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: lib/src/main/java/com/orange/ods/compose/component/button/OdsIconToggleButton.kt [hunk 1]
```diff
@@ -71,8 +72,13 @@ fun OdsIconToggleButton(
     ) {
         val iconTint by animateColorAsState(OdsTheme.colors.buttonToggleIconColor(displaySurface, checked))
         val backgroundAlpha by animateFloatAsState(if (checked) 0.12f else 0f)
+
         Box(
             modifier = Modifier
+                .border(
+                    width = 1.dp, color = OdsTheme.colors
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/button/OdsIconToggleButton.kt:79]: There should be a newline between `width` and `color` parameters.

File: demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonCustomizationState.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    val hasVariableWidth
+        get() = variableWidth.value
+}
+
+class ButtonToggleCustomizationState(
+    val toggleCount: MutableState<Int>,
+) {
+    companion object {
+        const val ToggleCountMin = 1
+        const val ToggleCountMax = 3
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonCustomizationState.kt:65]: In fact, I don't have any preference. I can change it to `MaxToggleCount` and `MinToggleCount` if you prefer. Maybe there are other max and min count variables in the code to change in this case.

File: demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonsContained.kt [hunk 3]
```diff
@@ -10,81 +10,130 @@
...
+                }
+                SwitchListItem(
+                    labelRes = R.string.component_element_icon,
+                    checked = leadingIcon,
+                )
+                SwitchListItem(
+                    labelRes = R.string.component_buttons_variable_width,
+                    checked = variableWidth,
+                )
+                SwitchListItem(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonsContained.kt:77]: I always have useless commas in my code :sweat_smile:

File: demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonsText.kt [hunk 4]
```diff
@@ -20,55 +27,113 @@ import androidx.compose.ui.res.stringResource
...
+            }
 
-    Spacer(modifier = Modifier.padding(top = dimensionResource(R.dimen.spacing_s)))
+            Column(
+                modifier = Modifier
+                    .verticalScroll(rememberScrollState())
+                    .padding(vertical = dimensionResource(id = R.dimen.screen_vertical_margin))
+            ) {
+                Title(titleRes, horizontalPadding = true)
+                TechnicalText(text = technicalText, withHorizontalPadding = true)
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonsText.kt:88]: Yes you're right!

[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ButtonsText.kt:88]: There is no style applied to the `OdsOutlinedButton` so I can't display this information but the screen is not homogeneous with the others...

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**